extmod/network_lwip.c

```diff
@@ -90,6 +100,241 @@ mp_obj_t mod_network_nic_ifconfig(struct netif *netif, size_t n_args, const mp_o
     }
 }
 
+mp_obj_t network_nic_config(struct netif *netif, size_t n_args, const mp_obj_t *args, mp_map_t *kwargs) {
+
+    if (kwargs->used == 0) {
+        // Get config value
+        if (n_args != 1) {
+            mp_raise_TypeError(MP_ERROR_TEXT("must query one param"));
+        }
+
+        switch (mp_obj_str_get_qstr(args[0])) {
+            case MP_QSTR_dhcp4: {
+                struct dhcp *dhcp = netif_dhcp_data(netif);
+                return dhcp != NULL && dhcp->state != DHCP_STATE_OFF ? mp_const_true : mp_const_false;
+            }
+            #if LWIP_IPV6_DHCP6
+            case MP_QSTR_dhcp6: {
+                struct dhcp6 *dhcp = netif_dhcp6_data(netif);
+                return dhcp != NULL && dhcp->state != DHCP6_STATE_OFF ? mp_const_true : mp_const_false;
+            }
+            #endif
+            #if LWIP_IPV6_AUTOCONFIG
+            case MP_QSTR_autoconf6: {
+                return netif->ip6_autoconfig_enabled ?  mp_const_true : mp_const_false;
+            }
+            #endif
+            case MP_QSTR_dns: {
+                const char *addr_str = ipaddr_ntoa(dns_getserver(0));
+                return mp_obj_new_str(addr_str, strlen(addr_str));
+            }
+            #if LWIP_VERSION_MAJOR >= 2
+            case MP_QSTR_prefer: {
+                return MP_OBJ_NEW_SMALL_INT(mp_mod_network_prefered_dns_ip);
+            }
+            #endif
+            case MP_QSTR_addr4: {
+                mp_obj_t tuple[2] = {
+                    netutils_format_ipv4_addr((uint8_t *)&netif->ip_addr, NETUTILS_BIG),
+                    netutils_format_ipv4_addr((uint8_t *)&netif->netmask, NETUTILS_BIG),
+                };
+                return mp_obj_new_tuple(2, tuple);
+            }
+            #if LWIP_IPV6
+            case MP_QSTR_addr6: {
+                mp_obj_t addrs[LWIP_IPV6_NUM_ADDRESSES];
+                size_t n_addrs = 0;
+                for (int i = 0; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
+                    if (ip6_addr_isvalid(netif_ip6_addr_state(netif, i))) {
+                        const char *addr_str = ipaddr_ntoa(netif_ip_addr6(netif, i));
+                        mp_obj_t tuple[4] = {
+                            mp_obj_new_str(addr_str, strlen(addr_str)),
+                            MP_OBJ_NEW_SMALL_INT(netif_ip6_addr_state(netif, i)),
```

Comment:
One of the API quirks elsewhere in micropython networking is not standardising the constant values across ports/implementations. (e.g. wireless security modes, or error codes).

So if e.g. we need to implement this on a non-LWIP NIC, we should implement a static mapping to defined constant values.

But in this case, are these values actually bool?

Reply:
`addr_state` is a flags variable, containing multiple bits. They are defined in `ip6_addr.h:283` following. They contain information if an address is valid, tentative, duplicated, and if it is preferred.

The other values `pref_life`/`valid_life` are numeric timers that are decreasing over timed.

Reply:
For the `pref_life`/`valid_life` values, I think it's fine to leave them as pure numbers, representing a count.  So they don't need constants.

For `addr_state`, that could have different values when/if `ipconfig()` is implemented on non-lwIP NICs.  So we either need to:
- provide constants in the network module itself
- standardise the bit fields/values of `addr_state` and enforce them to be the same for all implementations of all NICs, and then document these bits/values

I think the latter option is the best.  And then for the "standard" bits/values that we choose, it can be the same as what lwIP exposes.  That means that for now the code here can stay as-is, but if lwIP changes these values we will need to provide a translation mapping the new back to the old, so users can always rely on the same values working.  And then we need to document these lwIP values as our standard.